docs/pages/access-controls/guides/device-trust.mdx

```diff
@@ -0,0 +1,298 @@
+
+<Admonition type="tip" title="Enrollment">
+Device enrollment is key in correctly identifying a device throughout its
+lifetime. The more trusted the enrollment operator, the better the ongoing
+guarantees that the device itself is trustworthy.
+</Admonition>
+
+The tie between the device inventory and enrollment is a device enrollment
+token. A device enrollment token needs to be created by a device admin and sent
+to the person performing the enrollment ceremony. The token ties the actual
+device being enrolled to the expected device in the inventory, adding an extra
+layer of protection.
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- A signed and notarized `tsh` binary for enrollment.
+  [Download the macOS tsh installer](../../installation.mdx#macos).
+- A macOS device to register and enroll.
+
+## Step 1/2. Register a trusted device
+
+The `tctl` tool is used to manage the device inventory. A device admin is
+responsible for managing devices, adding new devices to the inventory and
+removing devices that are no longer in use.
+
+A role that grants permissions for the `device` resource is necessary to manage
+the inventory. Save the yaml below as `device-admin.yaml` and create it in your
+cluster:
+
+```yaml
+version: v6
+kind: role
+metadata:
+  name: device-admin
+spec:
+  allow:
+    rules:
+    - resources: ["device"]
+      verbs:
+      - create
+      - read
+      - list
+      - update
+      - delete
+      - create_enroll_token
+      - enroll
+```
+
+```code
+$ tctl create -f device-admin.yaml
+role 'device-admin' has been created
+```
+
+<Admonition type="tip" title="Editor Role">
+For clusters created using Teleport v12 or newer, the builtin role `editor` has
+the necessary permissions to manage devices. For production setups a separate
+role is still recommended.
+</Admonition>
+
+Note that in addition to the usual CRUD verbs (create, read, list, update and
+delete), we have also included `create_enroll_token` and `enroll`. The
+`create_enroll_token` verb is necessary to execute the `tctl devices enroll`
+command; `enroll` is necessary to execute `tsh device enroll`. Both are
+demonstrated in the following section.
+
+Make sure your user has the `device-admin` role:
+
+```code
+$ tctl edit users/alice
+```
+
+```diff
+kind: user
+metadata:
+  name: alice
+  # unrelated fields omitted.
+spec:
+  roles:
+  - access
++ - device-admin # add this line
+version: v2
+```
+
+Relogin to fetch updated certificates:
+
+```code
+$ tsh logout; tsh login --proxy=example.com --user=alice
+```
+
+Having the necessary permissions, let's add a device to the inventory. Replace
+the `$SERIAL` below with your macOS device serial number (visible under Apple ->
+"About This Mac" -> "Serial number", or use the command below).
+
+```code
+$ SERIAL="$(ioreg -c IOPlatformExpertDevice -d 2 | grep -i IOPlatformSerialNumber | awk -F'"' '{print $4}')"
+$ tctl devices add --os=macos --asset-tag="$SERIAL"
```

Comment:
It was at this point that my testing failed:

```
❯ tsh version
Teleport v12.0.0-alpha.2 git:api/v12.0.0-alpha.2-0-g3ebfce4a6a go1.19.5
Proxy version: 12.0.0-alpha.2

❯ tctl version
Teleport v12.0.0-alpha.2 git:api/v12.0.0-alpha.2-0-g3ebfce4a6a go1.19.5

❯ tctl get users/alex
kind: user
metadata:
  ...
  name: alex
spec:
  ...
  roles:
  - editor
  - access
  - device-admin
  status:
   ...
version: v2

❯ tctl get roles/device-admin
kind: role
metadata:
  ...
  name: device-admin
spec:
  allow:
    ...
    rules:
    - resources:
      - device
      verbs:
      - create
      - read
      - list
      - update
      - delete
      - create_enroll_token
      - enroll
  deny: {}
  options:
    ...
version: v5

❯ echo $SERIAL
REDACTED

❯ tctl devices add --os=macos --asset-tag="$SERIAL"
tctl: error: expected command but got "devices"

```

Reply:
You'll need the Enterprise version of `tctl` to do this part. We can make that clearer in the docs, or add to the troubleshooting section. WDYT?

Alternatively, we could move this code to OSS `tctl`, say for v12.1. @zmb3 for an opinion.